scripts/heratape_backup.py

```diff
@@ -0,0 +1,330 @@
+        local_paths = [entry["full_path_on_store"] for entry in local_instances["results"]]
+        local_obsids = [entry["obsid"] for entry in local_files["results"]]
+        
+        
+        logger.info(f'Found {len(local_names)} files, total size {total_size/1e12}TB')
+        logger.debug('first 10 duplicates')
+        for dup in find_dups(local_names)[:10]:
+            logger.debug(dup)
+        
+        # to build a record in heratape, we need information from Observations, Files, and Instances all zipped together
+        # append to the running list of data we're going to backup
+        # for each file, find one file instance, and find its matching observation info
+        for local_file in local_files["results"]:
+            #find a matching obs and instance
+            for local_instance in local_instances["results"]:
+                if local_instance["name"]==local_file["name"]: break
+            for local_obs in local_obses["results"]:
+                if local_file["obsid"]==local_obs["obsid"]: break
+            joined_results.append((local_instance,local_file,local_obs)) #IMPORTANT: This is where we are building the full backup list
+        jds_to_backup.append(jdtobackup)
+        continue #back to the top and see if we can add another day to this backup
+
+    logger.info('Moving on from data selection to tape writing!')
+    #example entry in "joined_results" which is the main data object that will be used to create db records
+    # and do the actual TAR
+    '''
+    ({'store_name': 'herastore01-8',
+      'store_ssh_host': 'heralib.aoc.nrao.edu',
+      'parent_dirs': '2459977',
+      'name': 'zen.2459977.21630.sum.autos.uvh5',
+      'deletion_policy': 'disallowed',
+      'full_path_on_store': '/export/hera/herastore01-8/2459977/zen.2459977.21630.sum.autos.uvh5'},
+     {'name': 'zen.2459977.21630.sum.autos.uvh5',
+      'type': 'uvh5',
+      'create_time': 1675303951,
+      'obsid': 1359306701,
+      'size': 19859900,
+      'md5': 'e910dffca62c8c859aacd873d2c14c6b'},
+     {'obsid': 1359306701,
+      'start_time_jd': 2459977.21624788,
+      'stop_time_jd': 2459977.21678797,
+      'start_lst_hr': 3.39516861968927,
+      'session_id': 1359303644})
+    '''
+    #turn our list of joined results into filenames, obsids, jds, and sizes
+    files = []
+    sizes = []
+    obsids = []
+    start_jds = []
+    for F in joined_results:
+        files.append(F[0]['full_path_on_store'])
+        sizes.append(F[1]['size'])
+        obsids.append(F[2]['obsid'])
+        start_jds.append(F[2]['start_time_jd'])
+    logger.info(f'Backing up JDS {"".join([str(jd) for jd in jds_to_backup])}. Nfiles = {len(files)}, Total Size = {np.round(np.sum(sizes)/1e12,1)}TB')
+    def update_tape_usage(tapes,testing=False):
+        #input: dict  from query_tape_usage and a session for heratape db
+        #output: input dict with added key: usage (sum in bytes known to heratape for each tape)
```

Comment:
need a proper docstring, numpy style. Also, why is this defined inside the while loop?